### PR TITLE
SD-1855 Fix package removal on archive purge

### PR DESCRIPTION
**Pull request by amagdas**
SD- 1858 Remove orphaned content from ingest

### CODE REVIEW

**Comment by amagdas** [comment]
@akintolga @marwoodandrew @mdhaman @sivakuna-aap @superdesk/developers

File: server/superdesk/io/commands/remove_expired_content.py [hunk 1]
```diff
@@ -58,31 +62,36 @@ def remove_expired_data(provider):
...
+        {'range': {'ingest.expiry': {'lte': date_to_str(utcnow())}}},
+        {'and': [{'missing': {'field': 'ingest.expiry'}},
+                 {'range': {'ingest.versioncreated': {'lte': date_to_str(expiration_date)}}}]},
     ]}
 
-    return superdesk.json.dumps(query)
+    if provider.get('_id'):
+        query['must'] = {'term': {'ingest.ingest_provider': str(provider.get('_id'))}}
+
+    if provider.get('exclude'):
```
Comment (akintolga) [inline server/superdesk/io/commands/remove_expired_content.py]: This "exclude" section is needed to find items with unknown providers. Is that right?

Comment (amagdas) [inline server/superdesk/io/commands/remove_expired_content.py]: Yes, basically, fetch all items which don't have a provider in the system

File: server/superdesk/io/commands/remove_expired_content.py [hunk 2]
```diff
@@ -30,15 +30,19 @@ class RemoveExpiredContent(superdesk.Command):
     )
 
     def run(self, provider_type=None):
-        for provider in superdesk.get_resource_service('ingest_providers').get(req=None, lookup={}):
+        providers = [p for p in superdesk.get_resource_service('ingest_providers').get(req=None, lookup={})]
+        self.remove_expired({'exclude': [str(p.get('_id')) for p in providers]})
```
Comment (akintolga) [inline server/superdesk/io/commands/remove_expired_content.py]: Can we do like this:
self.remove_expired({'exclude': [str(p.get('_id')) for p in providers if provider.get('type') != provider_type]})

Then maybe we don't need the line 35-37

**Comment by akintolga** [comment]
:+1:

**Comment by amagdas** [comment]
:watch:

**Comment by marwoodandrew** [comment]
:+1:

**Comment by amagdas** [comment]
:heart: :blue_heart: :yellow_heart:

**Pull request merged**